### ISSUE TITLE
Merge NoteActivity and EditNoteActivity

### ISSUE THREAD

**Issue by stefan-niedermann** [stefan-niedermann/nextcloud-notes#155]
These two Activities should be merged with the aim to use only one activity that is able to toggle between the EditView and the TextView.

Wanted behavior:

**Given** the user is in the EditNoteActivity
**When** the user clicks on the eye-icon
**Then** the EditView should be switched to the TextView
**And** the eye-icon should switch to the [eye-off-icon](https://materialdesignicons.com/icon/eye-off)

**When** the user clicks on the eye-off-icon
**Then** the TextView should be switched to the EditView
**And** the eye-off-icon should switch to the eye-icon

**Comment by korelstar** [comment]
I thought about this, too. An additional way to switch between the two views could be swiping the fragment to the left/right. I also thought about using tabs, but this would consume much space.

**Comment by stefan-niedermann** [comment]
Hm, no Tabs (needs too much space as you recognized yourself). Swiping? I don't know, gestures are very hard to discover and could be triggered accidentely. So *maybe* additional, but the primary way should be the eye/eye-off-icon.

**Comment by korelstar** [comment]
No problem. These were just some thoughts of mine.

**Comment by korelstar** [comment]
Another thought: It is saved whether *edit mode* or *preview mode* was active the last time, and when a (another or the same) note is opened the next time, the Activity switches automatically to the respective state. I envision to save the last state globally, not per note. This would make it more comfortable, when I want to read notes the most time and edit them seldom.

What do you think about this?

**Comment by stefan-niedermann** [comment]
@jancborchardt said
> [...] the edit mode should always be the default.
Though i think saving the state would make sense. *If* we save the state, we should do it globally indeed. Either way we will do it the exact same way like the server-side notes app will do to decrease confusion.
So Saving the toggle state depends on https://github.com/nextcloud/notes/issues/23 and could even be handeled in a seperate issue.

**Comment by korelstar** [comment]
It would be still the default, since for new users, the *edit mode* would be active. I don't know if @jancborchardt really meant that this applies to every *note opening*.

**Comment by stefan-niedermann** [comment]
Some christmas thoughts about how this could be implemented without a performance killer.
I thought we could add one `RxMDTextView` next to the `RxMDEditText` into the `activity_edit.xml`. Then we would have to implement a show/hide toggle mechanism. Problem: When user opens a note and everything runs in the main thread (and TextView *and* EditText are initialized) this could cause a weak performance.
Here is some pseudo-code:

```xml
on EditNoteActivity start:

start:

<async-task>
	if: Shared Preference/preview == false || unset
	then: initialize EditView and show
	and start:
		<async-task>
			initialize TextView
			when done: Show toggle button (X)
		</async-task>
		
	else: Shared Preference/preview == true
	then: initialize TextView and show
	and start:
		<async-task>
			initialize EditView
			when done: Show toggle button (X)
		</async-task>
</async-task>

on(X):
	set current text
	hide / show one of the both views
	set Shared Preference
```

Further thoughts: Since i don't know how android is handling the xml files internally i worry about that every xml tag is instantiated immediately, so maybe defining it in the xml is not the best option to implement this performant.

**Comment by schaarsc** [comment]
could this be a viable approach ... ???
a) convert NoteActivity into NotePreviewFragment
b) split EditNoteActivity into NoteEditFragment / NoteActivity(new)
c) in NoteActivity(new) configure the menu and handle toggle button by switching between NoteEditFragment <-> NotePreviewFragment
d) integrate CreateNoteActivity 

if you are interested, I'd try to prepare a PR ...

**Comment by jancborchardt** [comment]
I would advise to

- Have edit mode by default
- Save the view-state per-note, not globally. As if you usually never go into view mode, it’s confusing if that’s on after viewing one note.
- Make sure that always when you create a new note it’s in edit mode, not in view mode. ;)

**Comment by korelstar** [comment]
@schaarsc has implemented this in #242 👍
